### ISSUE TITLE
Ignore updated input files

### ISSUE THREAD

**Issue by jsgounot** [snakemake/snakemake#2328]
Let say we have a snakemake pipeline with `rule A -> rule B -> rule C` and rule A output is temporary. We run the pipeline once until rule B. We then rerun the pipeline entirely with `rerun-triggers mtime`. Snakemake will try to rerun everything even because input files for rule B are updated by job A. Is there a way to override this behavior without manually creating an empty file for rule  A and touching all output files? I feel like when running snakemake with `mtime` trigger, we should not expect snakemake to relaunch previous files because of updated input. Snakemake has a trigger named `input` that should include this case and not be triggered when it is not set.

**Comment by jsgounot** [comment]
Hi. Let me know if you think that might be a good idea, in this case I don't mind trying to generate a PR for this

**Comment by Hocnonsense** [comment]
I posted a [post on discord](https://discord.com/channels/753690260830945390/1129592542648868884/1129592542648868884) about this problem, and i'd copy my suggestion here. What do you think?

> Recently I'm dealing with a set of genomes, which are stored in gff format. Then I need to collect genes from all or some genomes. However, it's time consuming to annotate them, and it's space consuming to store all intermediate gene file of all genomes. `temp()` help nothing as if the file generated again, all things after it should be run again. 
> 
> I think one way to solute this problem is add a `definite` marker to generated gene file, indicating that output is already defined by rule and inputs, and nothing will happen the file is deleted and generated again. 
> 
> Moreover, we can generate a twin file for the gene file with a snakemake-format suffix. This file can just be a copy for related metadata in .snakemake/metadata folder. Then the file can be recognized and collected by git, similar to the version control provided by DVC. With a already existing metadata, a regenerated gene file will be checked with md5, file size, rule version or any other feature, and files after it will only be touched or ignored if it passed these checks.

**Comment by jsgounot** [comment]
Yep, this might work as well, it's just another way to see it. For my part, I think that the rerun-trigger keyword `input` is supposed to do this already, but does not. My problem here is that I feel like we might touch a snakemake behavior that Johannes has defined, and changing that will potentially impact well-defined processes. That's why I'm still waiting for an answer from a core contributor before going further.
